server.go

```diff
@@ -891,9 +891,6 @@ func (s *Server) processStreamingRPC(t transport.ServerTransport, stream *transp
 		trInfo:                trInfo,
 		statsHandler:          sh,
 	}
-	if ss.cp != nil {
```

Comment:
I don't think we want this change in this PR (?).  This buffer is still relevant and I think this is beneficial for streaming RPCs.

Reply:
This is actually needed, since now a SendMsg call can be made again before the previous msg was written on the wire. In such a case if we have a common buffer, we'll run into a race; concurrent access of the underlying buffer by encode and the writer goroutine.

Reply:
Interesting.  Did you run the benchmark for streaming large messages with only one concurrent call/connection?  (Does that benchmark do repeated sends on the stream or just one send and one receive?)  That is the only place where this change should show much impact, so it would be interesting to see how much, if at all, it was impacted.

Reply:
The streaming benchmark does sent multiple messages on a stream.
Run on master:

```
Stream-noTrace-latency_0s-kbps_0-MTU_0-maxConcurrentCalls_1-reqSize_1048576B-respSize_1048576B-Compressor_false
latency_0s-kbps_0-MTU_0-maxConcurrentCalls_1-reqSize_1048576B-respSize_1048576B-Compressor_false
    2116	   4727531 ns/op 10686814 B/op	    2390 allocs/op
Histogram (unit: ms)
Count: 2116  Min:   3.9  Max:   6.1  Avg: 4.73
------------------------------------------------------------
[      3.892775,       3.892776)     1    0.0%    0.0%  
[      3.892776,       3.892780)     0    0.0%    0.0%  
[      3.892780,       3.892801)     0    0.0%    0.0%  
[      3.892801,       3.892905)     0    0.0%    0.0%  
[      3.892905,       3.893434)     0    0.0%    0.0%  
[      3.893434,       3.896116)     0    0.0%    0.0%  
[      3.896116,       3.909704)     1    0.0%    0.1%  
[      3.909704,       3.978559)     7    0.3%    0.4%  
[      3.978559,       4.327462)   240   11.3%   11.8%  #
[      4.327462,       6.095440)  1866   88.2%  100.0%  #########
[      6.095440,            inf)     1    0.0%  100.0%  
```

On loopy writer:

```
Stream-noTrace-latency_0s-kbps_0-MTU_0-maxConcurrentCalls_1-reqSize_1048576B-respSize_1048576B-Compressor_false
latency_0s-kbps_0-MTU_0-maxConcurrentCalls_1-reqSize_1048576B-respSize_1048576B-Compressor_false
    2615	   3824685 ns/op 10669306 B/op	    1798 allocs/op
Histogram (unit: ms)
Count: 2615  Min:   3.2  Max:   4.7  Avg: 3.82
------------------------------------------------------------
[      3.182852,       3.182853)     1    0.0%    0.0%  
[      3.182853,       3.182857)     0    0.0%    0.0%  
[      3.182857,       3.182876)     0    0.0%    0.0%  
[      3.182876,       3.182968)     0    0.0%    0.0%  
[      3.182968,       3.183415)     0    0.0%    0.0%  
[      3.183415,       3.185592)     0    0.0%    0.0%  
[      3.185592,       3.196198)     2    0.1%    0.1%  
[      3.196198,       3.247853)    15    0.6%    0.7%  
[      3.247853,       3.499428)   283   10.8%   11.5%  #
[      3.499428,       4.724684)  2313   88.5%  100.0%  #########
[      4.724684,            inf)     1    0.0%  100.0%  
```